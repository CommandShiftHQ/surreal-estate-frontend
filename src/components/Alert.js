import React from "react";
import PropTypes from "prop-types";

import "../styles/Alert.css";

const Alert = ({ message, success }) => {
  if (!message) return null;

  return (
    <p className={`Alert alert-${success ? "success" : "error"}`}>{message}</p>
  );
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
