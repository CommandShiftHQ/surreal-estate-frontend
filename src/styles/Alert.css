.Alert {
  padding: 5px;
  margin-bottom: 5px;
}

.alert-success {
  border: 1px solid lightgreen;
  background-color: lightgreen;
  color: green;
}

.alert-error {
  border: 1px solid lightpink;
  background-color: lightpink;
  color: red;
}
