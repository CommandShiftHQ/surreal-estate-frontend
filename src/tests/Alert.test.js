import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

test("displays an error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(asFragment()).toMatchSnapshot();
  expect(getByText(/Error/).textContent).toBe("Error!");
});

test("displays a success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!!" success />
  );

  expect(asFragment()).toMatchSnapshot();
  expect(getByText(/Success/).textContent).toBe("Success!!!!");
});
