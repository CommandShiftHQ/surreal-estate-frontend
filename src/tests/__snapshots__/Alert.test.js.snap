// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a success message 1`] = `
<DocumentFragment>
  <p
    class="Alert alert-success"
  >
    Success!!!!
  </p>
</DocumentFragment>
`;

exports[`displays an error message 1`] = `
<DocumentFragment>
  <p
    class="Alert alert-error"
  >
    Error!
  </p>
</DocumentFragment>
`;
